From 000e35e398d9bf8fc4429093a961d3020788e994 Mon Sep 17 00:00:00 2001
From: Darshan Prajapati <darshan.prajapati@einfochips.com>
Date: Wed, 5 Jun 2024 12:21:47 +0530
Subject: [PATCH 4/5] lib: sbi: Set tor size

Set reg field of sbi_domain_memregion to set end size of TOR region

Upstream-Status: Pending

Signed-off-by: Darshan Prajapati <darshan.prajapati@einfochips.com>
---
 lib/sbi/sbi_domain.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/lib/sbi/sbi_domain.c b/lib/sbi/sbi_domain.c
index 4e9f742..632c407 100644
--- a/lib/sbi/sbi_domain.c
+++ b/lib/sbi/sbi_domain.c
@@ -85,7 +85,8 @@ ulong sbi_domain_get_assigned_hartmask(const struct sbi_domain *dom,
 void sbi_domain_memregion_init(unsigned long addr,
 				unsigned long size,
 				unsigned long flags,
-				struct sbi_domain_memregion *reg)
+				struct sbi_domain_memregion *reg,
+				unsigned long tor)
 {
 	unsigned long base = 0, order;
 
@@ -108,6 +109,7 @@ void sbi_domain_memregion_init(unsigned long addr,
 		reg->base = base;
 		reg->order = order;
 		reg->flags = flags;
+		reg->tor = tor;
 	}
 }
 
@@ -442,8 +444,12 @@ void sbi_domain_dump(const struct sbi_domain *dom, const char *suffix)
 	i = 0;
 	sbi_domain_for_each_memregion(dom, reg) {
 		rstart = reg->base;
-		rend = (reg->order < __riscv_xlen) ?
-			rstart + ((1UL << reg->order) - 1) : -1UL;
+		if (!reg->tor) {
+			rend = (reg->order < __riscv_xlen) ?
+				rstart + ((1UL << reg->order) - 1) : -1UL;
+		}else{
+			rend = rstart + reg->tor - 1;
+		}
 
 		sbi_printf("Domain%d Region%02d    %s: 0x%" PRILX "-0x%" PRILX " ",
 			   dom->index, i, suffix, rstart, rend);
@@ -800,7 +806,7 @@ int sbi_domain_init(struct sbi_scratch *scratch, u32 cold_hartid)
 				  (SBI_DOMAIN_MEMREGION_SU_READABLE |
 				   SBI_DOMAIN_MEMREGION_SU_WRITABLE |
 				   SBI_DOMAIN_MEMREGION_SU_EXECUTABLE),
-				  &root_memregs[root_memregs_count++]);
+				  &root_memregs[root_memregs_count++],0);
 
 	/* Root domain memory region end */
 	root_memregs[root_memregs_count].order = 0;
-- 
2.25.1

